src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -78,14 +98,16 @@ public static void generateReposReport(List<RepoConfiguration> configs, String o
             AuthorshipSummary authorshipSummary = AuthorshipReporter.generateAuthorshipSummary(config);
             generateIndividualRepoReport(commitSummary, authorshipSummary, repoReportDirectory.toString());
 
-            try {
-                FileUtil.deleteDirectory(FileUtil.REPOS_ADDRESS);
-            } catch (IOException ioe) {
-                logger.log(Level.WARNING, "Error deleting report directory.", ioe);
+            if (shouldDeleteDirectory) {
+                FileUtil.deleteDirectory(config.getRepoRoot());
```

Comment:
as mentioned earlier, clean up can be on the cloner side.
It is easier for the cloner to determine whether the directory can be deleted.
E.g. if the array slot is going to be used to clone new repo, old one can be deleted. 
when all the analyzing is done, can call for clean up too.